x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/TransportNodeDeprecationCheckAction.java

```diff
@@ -123,8 +132,49 @@ protected NodesDeprecationCheckResponse newResponse(
             nodeSettingsChecks,
             (c) -> c.apply(filteredNodeSettings, pluginsService.info(), filteredClusterState, licenseState)
         );
-
+        addDiskUsageWarnings(issues, filteredNodeSettings, filteredClusterState.metadata().settings());
         return new NodesDeprecationCheckAction.NodeResponse(transportService.getLocalNode(), issues);
     }
 
+    private void addDiskUsageWarnings(List<DeprecationIssue> issues, Settings nodeSettings, Settings clusterSettings) {
+        DiskUsage usage = clusterInfoService.getClusterInfo().getNodeMostAvailableDiskUsages().get(transportService.getLocalNode().getId());
+        if (usage != null) {
+            long freeBytes = usage.getFreeBytes();
+            double freeDiskPercentage = usage.getFreeDiskAsPercentage();
+            final boolean emitWarning;
+            if (exceedsLowWatermark(nodeSettings, freeBytes, freeDiskPercentage)
+                || exceedsLowWatermark(clusterSettings, freeBytes, freeDiskPercentage)) {
+                emitWarning = true;
```

Comment:
The use of this boolean seems a little strange. Why do we have it when it could be:

```java
if (exceedsLowWatermark(...) || exceedsLowWatermark(...)) {
    issues.add(...);
}
```

and then we don't need an else branch, since it's not used anywhere?

Reply:
Ohh, that's because I initially had some more complex logic in here for outputting a meta tag indicating that the dynamic property could be removed... until I realized that that made no sense. I'll clean this up.